### PR TITLE
Bookmark model

### PR DESCRIPTION
**Pull request by gamma-draconis**
Added the bookmark model. Rails g only added bookmark.rb and bookmark_spec.rb.

### CODE REVIEW

File: app/models/bookmark.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Bookmark
+    include Mongoid::Document
+    field :url, :type => String
+    field :title, :type => String
+    field :description, :type => String
+    field :tags, :type => Array
+    field :privacy, :type => String
+    field :userID, :type => String
```
Comment (cesy) [inline app/models/bookmark.rb]: Why don't we use belongs_to :user here, as we already have a user model?

File: app/models/bookmark.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Bookmark
+    include Mongoid::Document
+    field :url, :type => String
+    field :title, :type => String
+    field :description, :type => String
+    field :tags, :type => Array
+    field :privacy, :type => String
```
Comment (cesy) [inline app/models/bookmark.rb]: I'd recommend calling it "private" or "public" and making it a Boolean - any network choice for showing certain bookmarks to friends only can be adapted later, but making this a string will give us a lot of headaches.

File: app/models/bookmark.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    field :userID, :type => String
+    field :timestamp, :type => Time
+
+    validates :url, :presence => true,
+                    :length => {:minimum => 2}
+
+    validates :title, :presence => true,
+                      :length => {:minimum => 2}
```
Comment (cesy) [inline app/models/bookmark.rb]: We should probably also have a maximum for this, and include the error messages (e.g. ":too_short => 'must be at least 2 characters long'")

**Comment by cesy** [comment]
This looks good to me otherwise. Our next priority should probably then be to get the specs and cukes running.

**Comment by anathomical** [comment]
I think we also want to white list accessible attributes (http://stackoverflow.com/questions/2652907/what-is-difference-between-attr-accessibleattributes-attr-protectedattrib).  For security.

And while there's no code change necessary for this, what do people think of using a user's email address as their userID?  If we're okay with that, I'd suggest tossing in a comment to that effect, just for completeness sake.

**Comment by gamma-draconis** [comment]
Bookmark CRUD will get right on the accessible attributes which won't take
too much time. And as for the email as userID: I don't really mind either
way. (Also, I'll get on commenting the code.) :)

--Jase

On Sat, Mar 24, 2012 at 12:53 PM, Anathomical <
reply@reply.github.com

**Comment by gamma-draconis** [comment]
So I made url, title, description, tags, and is_private all attr_accessible.

**Pull request merged**